### PR TITLE
Feat/8891 ra graphql upgrade

### PR DESCRIPTION
**Pull request by morhag90**
Fixes: #8891

## PR Details

Update admin-ui package to support ra-graphql update. 

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/data-service-generator/src/admin/package-json/package.json [hunk 1]
```diff
@@ -3,54 +3,40 @@
...
+    "@types/react": "^18.3.3",
+    "@types/react-dom": "^18.3.0",
+    "@vitejs/plugin-react": "^4.0.1",
+    "eslint": "^8.43.0",
+    "eslint-config-prettier": "^8.8.0",
+    "eslint-plugin-react": "^7.32.2",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "prettier": "^2.8.8",
+    "typescript": "^5.1.6",
+    "vite": "^4.3.9",
```
Comment (mulygottlieb) [inline packages/data-service-generator/src/admin/package-json/package.json:38]: Is vite only needed for dev time scripts and not used in runtime by react-admin 5?

File: packages/data-service-generator/src/admin/static/src/vite-env.d.ts
Comment (mulygottlieb) [inline packages/data-service-generator/src/admin/static/src/vite-env.d.ts:1]: Do we need this file?

File: packages/data-service-generator/src/admin/package-json/package.json (hunk 1 above)
Comment (yuval-hazaz) [reply packages/data-service-generator/src/admin/package-json/package.json:38]: yes

File: packages/data-service-generator/src/admin/static/src/vite-env.d.ts
Comment (yuval-hazaz) [reply packages/data-service-generator/src/admin/static/src/vite-env.d.ts:1]: /// <reference types="vite/client" />: Tells TypeScript to include Vite's client-side type definitions.

**Review by yuval-hazaz — APPROVED** [review]

**Review by mulygottlieb — APPROVED** [review]

**Review by mulygottlieb — APPROVED** [review]

**Pull request merged**